CRM/Contact/DAO/RelationshipVortex.php

```diff
@@ -0,0 +1,328 @@
+      Civi::$statics[__CLASS__]['links'][] = new CRM_Core_Reference_Basic(self::getTableName(), 'relationship_id', 'civicrm_relationship', 'id');
+      Civi::$statics[__CLASS__]['links'][] = new CRM_Core_Reference_Basic(self::getTableName(), 'relationship_type_id', 'civicrm_relationship_type', 'id');
+      Civi::$statics[__CLASS__]['links'][] = new CRM_Core_Reference_Basic(self::getTableName(), 'near_contact_id', 'civicrm_contact', 'id');
+      Civi::$statics[__CLASS__]['links'][] = new CRM_Core_Reference_Basic(self::getTableName(), 'far_contact_id', 'civicrm_contact', 'id');
+      CRM_Core_DAO_AllCoreTables::invoke(__CLASS__, 'links_callback', Civi::$statics[__CLASS__]['links']);
+    }
+    return Civi::$statics[__CLASS__]['links'];
+  }
+
+  /**
+   * Returns all the column names of this table
+   *
+   * @return array
+   */
+  public static function &fields() {
+    if (!isset(Civi::$statics[__CLASS__]['fields'])) {
+      Civi::$statics[__CLASS__]['fields'] = [
+        'id' => [
+          'name' => 'id',
+          'type' => CRM_Utils_Type::T_INT,
+          'title' => ts('Relationship ID'),
+          'description' => ts('Relationship ID'),
+          'required' => TRUE,
+          'where' => 'civicrm_relationship_vtx.id',
+          'table_name' => 'civicrm_relationship_vtx',
+          'entity' => 'RelationshipVortex',
+          'bao' => 'CRM_Contact_BAO_RelationshipVortex',
+          'localizable' => 0,
+        ],
+        'relationship_id' => [
+          'name' => 'relationship_id',
+          'type' => CRM_Utils_Type::T_INT,
+          'title' => ts('Relationship'),
+          'description' => ts('id of the relationship'),
+          'required' => TRUE,
+          'where' => 'civicrm_relationship_vtx.relationship_id',
+          'table_name' => 'civicrm_relationship_vtx',
+          'entity' => 'RelationshipVortex',
+          'bao' => 'CRM_Contact_BAO_RelationshipVortex',
+          'localizable' => 0,
+          'FKClassName' => 'CRM_Contact_DAO_Relationship',
+          'html' => [
+            'type' => 'Select',
+          ],
+        ],
+        'relationship_type_id' => [
+          'name' => 'relationship_type_id',
+          'type' => CRM_Utils_Type::T_INT,
+          'title' => ts('Relationship Type'),
+          'description' => ts('id of the relationship'),
+          'required' => TRUE,
+          'where' => 'civicrm_relationship_vtx.relationship_type_id',
+          'table_name' => 'civicrm_relationship_vtx',
+          'entity' => 'RelationshipVortex',
+          'bao' => 'CRM_Contact_BAO_RelationshipVortex',
+          'localizable' => 0,
+          'FKClassName' => 'CRM_Contact_DAO_RelationshipType',
+          'html' => [
+            'type' => 'Select',
+          ],
+        ],
+        'orientation' => [
+          'name' => 'orientation',
+          'type' => CRM_Utils_Type::T_INT,
+          'title' => ts('Orientation (A/B or B/A)'),
```

Comment:
Existing code uses the convention `"a_b"` or `"b_a"` so it would be nice to stick with that and not have to reinvent it as `"A/B"` `"B/A"`

Reply:
Sure, I'm pushing an update to use `a_b`/`b_a` instead of 0/1.